### PR TITLE
test: add different language tests and modify some cases

### LINKED ISSUE #36396: [Bug]: [Nightly] Insert inconsistent will not raise exception
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 89397d1
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):    rocksmq
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Insert inconsistent will not raise exception.
```
[pytest : test]     @pytest.mark.tags(CaseLabel.L2)
[pytest : test]     def test_insert_inconsistent_data(self):
[pytest : test]         """
[pytest : test]         target: test insert with inconsistent data
[pytest : test]         method: insert with data that same field has different type data
[pytest : test]         expected: raise exception
[pytest : test]         """
[pytest : test]         c_name = cf.gen_unique_str(prefix)
[pytest : test]         collection_w = self.init_collection_wrap(name=c_name)
[pytest : test]         data = cf.gen_default_list_data(nb=100)
[pytest : test]         data[0][1] = 1.0
[pytest : test]         error = {ct.err_code: 999,
[pytest : test]                  ct.err_msg: "The Input data type is inconsistent with defined schema, {%s} field should be a int64, "
[pytest : test]                              "but got a {<class 'int'>} instead." % ct.default_int64_field_name}
[pytest : test] >       collection_w.insert(data, check_task=CheckTasks.err_res, check_items=error)

[pytest : test] [2024-09-19 19:22:17 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [[0      0
[pytest : test] 1      1
[pytest : test] 2      2
[pytest : test] 3      3
[pytest : test] 4      4
[pytest : test]       ..
[pytest : test] 95    95
[pytest : test] 96    96
[pytest : test] 97    97
[pytest : test] 98    98
[pytest : test] 99    99
[pytest : test] Length: 100, dtype: int64, 0      0.0
[pytest : test] 1      1.0
[pytest : test] 2      2.0
[pytest : test] 3      3.0
[pytest : test] 4      4.0
[pytest : test]       ... 
[pytest : test] 95    95.0
[pytest : test] 96    96.0
[pytest : test] 97    97.0
[pytest : test] 98    98.0
[pytest : test] 99    99.0
[pytest : test] Length: 100, dtype: float32, 0      0
[pytest : test] 1      1
[pytest : test] 2   ......, kwargs: {'timeout': 180} (api_request.py:62)
[pytest : test] [2024-09-19 19:22:17 - DEBUG - ci_test]: (api_response) : (insert count: 100, delete count: 0, upsert count: 0, timestamp: 452663374759067660, success count: 100, err count: 0  (api_request.py:37)
```

### Expected Behavior

report error

### Steps To Reproduce

_No response_

### Milvus Log

1. link: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI(new)/detail/master/122/pipeline/163/
2. log: [artifacts-milvus-standalone-ms-master-122-py-n-122-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI(new)/job/master/122/artifact/artifacts-milvus-standalone-ms-master-122-py-n-122-e2e-logs.tar.gz)
3. failed case: [gw2] [ 23%] FAILED testcases/test_insert.py::TestInsertParams::test_insert_inconsistent_data 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by NicoYuan1986**
fix: #36396

### CODE REVIEW

File: tests/python_client/testcases/test_search.py [hunk 1]
```diff
@@ -5926,6 +5932,48 @@ def test_search_string_field_not_primary_is_empty(self, _async):
...
+    def test_search_string_different_language(self):
+        """
+        target: test search with string expr using different language
+        method: create collection and insert data
+                create index and collection load
+                collection search uses string expr in string field
+        expected: Search successfully
+        """
+        # 1. initialize with data
+        _async = random.choice([True, False])
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:5945]: good point to make a balance between case coverage and case numbers.

**Comment by binbinlv** [comment]
please add the results check between inserted string and the results from query/search for different languages, maybe in your next pr is OK for me.

**Comment by NicoYuan1986** [comment]
> please add the results check between inserted string and the results from query/search for different languages, maybe in your next pr is OK for me.

No problem!

**Pull request merged**